agnpy/radiative_process/tests/test_time_evolution.py

```diff
@@ -0,0 +1,301 @@
+        ssc = SynchrotronSelfCompton(blob)
+
+        time = 6 * u.s
+        intervals = 300
+
+        electrons_before = n_e.integrate()
+
+        TimeEvolution(blob, time, synchrotron_loss(synch)).eval_with_fixed_intervals(
+                                                      intervals_count=intervals, max_change_per_interval=0.5)
+        electrons_after_sync = blob.n_e.integrate()
+
+        blob.n_e = n_e
+        TimeEvolution(blob, time, ssc_thomson_limit_loss(ssc)).eval_with_fixed_intervals(
+                                                      intervals_count=intervals, max_change_per_interval=0.5)
+        electrons_after_ssc_th = blob.n_e.integrate()
+
+        blob.n_e = n_e
+        TimeEvolution(blob, time, ssc_loss(ssc)).eval_with_fixed_intervals(
+                                                      intervals_count=intervals, max_change_per_interval=0.5)
+        electrons_after_ssc = blob.n_e.integrate()
+
+        assert u.isclose(electrons_before, electrons_after_sync, rtol=0.005)
+        assert u.isclose(electrons_before, electrons_after_ssc_th, rtol=0.005)
+        assert u.isclose(electrons_before, electrons_after_ssc, rtol=0.005)
+
+    def test_ssc_losses_in_thompson_limit(self):
+        r_b = 1e16 * u.cm
+        n_e = PowerLaw.from_total_energy(
+            1e30 * u.erg,
+            4 / 3 * np.pi * r_b ** 3,
+            p=2.8,
+            gamma_min=1e1,
+            gamma_max=1e3,
+            mass=m_e,
+        )
+
+        blob = Blob(r_b, n_e=n_e)
+        ssc = SynchrotronSelfCompton(blob)
+
+        thomson = ssc.electron_energy_loss_rate_thomson(np.logspace(1, 3, 200)).to("erg/s")
+        ssc = ssc.electron_energy_loss_rate(np.logspace(1, 3, 200)).to("erg/s")
+
+        assert u.allclose(thomson, ssc, 0.01)
+
+    def test_compare_numerical_results_with_analytical_calculation_ssc_thomson(self):
+        """Test time evolution of spectral electron density for SSC energy losses in Thomson range.
+         Similar to the corresponding test for Synchrotron losses."""
+        time = 1e26 * u.s # energy losses are very low, of the order of 1e-33 erg/s,
+                          # so we need to multiply them by a long time to observe any change in the electron distribution
+        steps = 1
+        p = 0.5
+        r_b = 1e16 * u.cm
+        gamma_min = 1e3
+        gamma_max = 1e4
+        n_e = PowerLaw.from_total_energy(
+            1e20 * u.erg,
+            4 / 3 * np.pi * r_b ** 3,
+            p=p,
+            gamma_min=gamma_min,
+            gamma_max=gamma_max,
+            mass=m_e,
+        )
+        k = n_e.k
+        blob = Blob(r_b, n_e=n_e, B=10*u.G)
+        initial_gammas = blob.gamma_e
+
+        ssc = SynchrotronSelfCompton(blob)
+        TimeEvolution(blob, time, ssc_loss(ssc)).eval_with_fixed_intervals(intervals_count=steps)
+        evaluated_n_e: InterpolatedDistribution = blob.n_e
+
+        def gamma_before(prefactor, gamma_after_time, before_time):
```

Comment:
this function looks very much like gamma_before_time defined at the beginning

Reply:
yes, a leftover from an earlier versions, removed now